src/vcpkg/vcpkgpaths.cpp

```diff
@@ -971,8 +971,16 @@ namespace vcpkg
         }
         return ret;
     }
+
     std::string VcpkgPaths::get_current_git_sha_baseline_message() const
     {
+        const auto& git_config = git_builtin_config();
+        auto maybe_shallow_clone = is_shallow_clone(git_config).get();
+        if (maybe_shallow_clone && *maybe_shallow_clone)
```

Comment:
This is a use after free; the pointer returned by get() points to the temporary ExpectedL in this expression.

I think you should add:

```
        template<class... Args>
        T value_or(Args&&... or_args) &&
        {
            auto maybe_value = m_t.get();
            if (maybe_value)
            {
                return std::move(*maybe_value);
            }

            return T(std::forward<Args>(or_args)...);
        }

        template<class... Args>
        T value_or(Args&&... or_args) const&
        {
            auto maybe_value = m_t.get();
            if (maybe_value)
            {
                return *maybe_value;
            }

            return T(std::forward<Args>(or_args)...);
        }
```

to `expected.h` right above the `value_or_exit` overloads, then change this to:

```suggestion
        if (is_shallow_clone(git_config).value_or(false))
```